patches/minecraft/net/minecraft/data/recipes/CraftingRecipeBuilder.java.patch

```diff
@@ -0,0 +1,27 @@
+--- a/net/minecraft/data/recipes/CraftingRecipeBuilder.java
++++ b/net/minecraft/data/recipes/CraftingRecipeBuilder.java
+@@ -5,23 +_,7 @@
+ 
+ public abstract class CraftingRecipeBuilder {
+    protected static CraftingBookCategory m_245179_(RecipeCategory p_250736_) {
+-      CraftingBookCategory craftingbookcategory;
+-      switch (p_250736_) {
+-         case BUILDING_BLOCKS:
+-            craftingbookcategory = CraftingBookCategory.BUILDING;
+-            break;
+-         case TOOLS:
+-         case COMBAT:
+-            craftingbookcategory = CraftingBookCategory.EQUIPMENT;
+-            break;
+-         case REDSTONE:
+-            craftingbookcategory = CraftingBookCategory.REDSTONE;
+-            break;
+-         default:
+-            craftingbookcategory = CraftingBookCategory.MISC;
+-      }
+-
+-      return craftingbookcategory;
++      return p_250736_.getCraftingBookCategory();
```

Comment:
No, this is not how you replace a method. If you wanted to do something for this, either replace the return statement or, more reasonably the default statement with a forge hook that maps recipe categories to their crafting book category.